### PR TITLE
Final changes before merging back to master.

### PR DESCRIPTION
**Pull request by aomarks**
This should be the final set of changes before merging the analyzer branch back to master. Sorry for the giant grab bag, you may want to look at commits individually.

Highlights:

- Test coverage for all the child components.
- No more `paper-*` dependencies. Inlined colors and added `iron-doc-font-*` mixins with an optional `default-theme.html`.
- Added a `path` property that defaults to an `iron-location` so that the currently displayed element can be controlled directly without being tied to the URL (good for testing and whatever other custom routing might be wanted).
- Updated documentation.
- Remove feature that loads JSON as child content. It's not being used by anyone AFAIK.

### CODE REVIEW

File: iron-doc-function.html [hunk 1]
```diff
@@ -23,31 +23,29 @@
...
     </style>
 
-    <div id="transitionMask">
-      <div id="signature">
-        <a href$="#[[anchorId]]" class="name deeplink">[[descriptor.name]]</a>(<span class="params code"><span>[[_paramText]]</span></span>)<!--
-        --><span class="return" hidden$="[[!descriptor.return]]">: <span class="type code">[[descriptor.return.type]]</span></span>
-      </div>
+    <div id="signature">
+      <a href$="#[[anchorId]]" class="name deeplink">[[descriptor.name]]</a>(<span class="params code"><span>[[_paramText]]</span></span>)<!--
+      --><span class="return" hidden$="[[!descriptor.return]]">: <span class="type code">[[descriptor.return.type]]</span></span>
```
Comment (samuelli) [inline iron-doc-function.html:28]: Remove empty comment?

File: iron-doc-viewer.html [hunk 2]
```diff
@@ -85,30 +107,78 @@
...
 
-        _dataChanged: function(descriptor, routePath) {
+        _pathChanged: function(path) {
+          this._pathSetExternally = path && !this._ourPathChanging;
+        },
+
+        _ourPathChanged: function(ourPath) {
+          if (ourPath) {
+            this._ourPathChanging = true;
+            this.path = ourPath;
```
Comment (samuelli) [inline iron-doc-viewer.html]: Not sure if there's a better way but looks kinda messy. Maybe a comment at least?

**Review by samuelli — APPROVED** [review]

File: iron-doc-function.html (hunk 1 above)
Comment (aomarks) [reply iron-doc-function.html:28]: I think it's there to collapse the space.

File: iron-doc-viewer.html (hunk 2 above)
Comment (aomarks) [reply iron-doc-viewer.html]: Done. I may actually rip this feature out and require something more explicit (caller just provides the route directly or something).

**Pull request merged**